hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BPServiceActor.java

```diff
@@ -258,7 +258,9 @@ NamespaceInfo retrieveNamespaceInfo() throws IOException {
     while (shouldRun()) {
       try {
         nsInfo = bpNamenode.versionRequest();
-        LOG.debug(this + " received versionRequest response: " + nsInfo);
+        if (LOG.isDebugEnabled()) {
+          LOG.debug(this + " received versionRequest response: " + nsInfo);
+        }
```

Comment:
I think if we change this to
```
        LOG.debug("{} received versionRequest response: {}", this, nsInfo);
```

in this case as well, ``toString()`` won't be invoked & we can get away without having this ``isDebugEnabled`` gaurd

Reply:
Thank you for your advice, this way can also achieve the same effect.